odoo/models.py

```diff
@@ -1761,15 +1763,21 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
         :return: list of pairs ``(id, display_name)`` for all matching records.
         """
         ids = self._name_search(name, args, operator, limit=limit, order=self._order)
-
+        extra_fields = [field for field in extra_fields if field in self._fields] if bool(extra_fields) else []
         if isinstance(ids, Query):
-            records = self._fetch_query(ids, self._determine_fields_to_fetch(['display_name']))
+            records = self._fetch_query(ids, self._determine_fields_to_fetch(['display_name', *extra_fields]))
         else:
             # Some override of `_name_search` return list of ids.
             records = self.browse(ids)
-            records.fetch(['display_name'])
-
-        return [(record.id, record.display_name) for record in records.sudo()]
+            records.fetch(['display_name', *extra_fields])
+        res = []
+        for record in records.sudo():
+            extra = ({field: record[field] for field in extra_fields if field in self._name_search_allowed_extra_fields})
+            if bool(extra):
```

Comment:
this doesn't look correct, what if the value of extra is a boolean/empty string/0 number, etc?

Reply:
I misunderstood the extra (dict) from a specific field value.

Reply:
casting to bool is not necessary when used inside conditions though